infrastructure/resource_group_level/modules/function-acbs.bicep

```diff
@@ -0,0 +1,172 @@
+param location string
+param environment string
+param containerRegistryName string
+param appServicePlanEgressSubnetId string
+param appServicePlanId string
+param privateEndpointsSubnetId string
+param storageAccountName string
+
+
+// These values are taken from GitHub secrets
+@secure()
+param secureSettings object = {
+  MULESOFT_API_KEY: 'test-value'
+  MULESOFT_API_SECRET: 'test-value'
+  MULESOFT_API_UKEF_TF_EA_URL: 'test-value'
+  APIM_MDM_KEY: 'test-value'
+  APIM_MDM_URL: 'test-value'
+  APIM_MDM_VALUE: 'test-value' // different in staging and dev
+}
+
+// These values are taken from an export of Configuration on Dev
+// TODO:FN-419 Add to GitHub secrets?
+@secure()
+param additionalSecureSettings object = {
+  DOCKER_REGISTRY_SERVER_PASSWORD: 'test-value'  // different in staging and dev
+  MACHINEKEY_DecryptionKey: 'test-value' // different in staging and dev
+  MULESOFT_API_UKEF_MDM_EA_KEY: 'test-value'
+  MULESOFT_API_UKEF_MDM_EA_SECRET: 'test-value'
+  MULESOFT_API_UKEF_MDM_EA_URL: 'test-value' // different in staging and dev
+}
+
+
+var dockerImageName = '${containerRegistryName}.azurecr.io/azure-function-acbs:${environment}'
+
+var dockerRegistryServerUsername = 'tfs${environment}'
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' existing = {
+  name: storageAccountName
+}
+
+var storageAccountKey = storageAccount.listKeys().keys[0].value
+
+// These values are hardcoded in the CLI scripts
+var settings = {
+  FUNCTIONS_WORKER_RUNTIME: 'node'
+  // TODO:FN-419 consider making this a parameter
+  WEBSITE_DNS_SERVER: '168.63.129.16'
+  WEBSITE_VNET_ROUTE_ALL: '1'
+}
+
+// These values are taken from an export of Configuration on Dev
+var additionalSettings = {
```

Comment:
I'm not totally happy with the Azure convention of `param`, `var`, `resource` ordering here as this var has a dependency on the applicationInsights resource below. I suppose it is ok.